### PR TITLE
Validate that @ConfigMapping annotation can only be placed in interfaces

### LINKED ISSUE #424: Validate that @ConfigMapping annotation can only be placed in interfaces
@ConfigMaping can be used only for interface and not for class.

 * Using interface is valid:
 
```java
@ConfigMapping(prefix = "server")
public interface Server {

}
```

* Using class should higlight the Server as an error:
 
```java
@ConfigMapping(prefix = "server")
public class Server {

}
```

Here the Quarkus stack trace when we try to annotated a Java class with ConfigMapping:

```

java.lang.IllegalStateException: SRCFG00043: The @ConfigMapping annotation can only be placed in interfaces, class org.acme.Server is a class
	at io.smallrye.config.ConfigMappingLoader.validateAnnotations(ConfigMappingLoader.java:104)
	at io.smallrye.config.ConfigMappingLoader.getConfigMappingClass(ConfigMappingLoader.java:44)
	at io.smallrye.config.ConfigMappingLoader.getConfigMappingInterface(ConfigMappingLoader.java:40)
	at io.smallrye.config.ConfigMappingLoader.getConfigMappingsMetadata(ConfigMappingLoader.java:34)
	at io.quarkus.deployment.configuration.ConfigMappingUtils.generateConfigClasses(ConfigMappingUtils.java:79)
	at io.quarkus.arc.deployment.ConfigBuildStep.generateConfigClasses(ConfigBuildStep.java:246)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Resulted in: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ConfigBuildStep#generateConfigClasses threw an exception: java.lang.IllegalStateException: SRCFG00043: The @ConfigMapping annotation can only be placed in interfaces, class org.acme.Server is a class
	at io.smallrye.config.ConfigMappingLoader.validateAnnotations(ConfigMappingLoader.java:104)
	at io.smallrye.config.ConfigMappingLoader.getConfigMappingClass(ConfigMappingLoader.java:44)
	at io.smallrye.config.ConfigMappingLoader.getConfigMappingInterface(ConfigMappingLoader.java:40)
	at io.smallrye.config.ConfigMappingLoader.getConfigMappingsMetadata(ConfigMappingLoader.java:34)
	at io.quarkus.deployment.configuration.ConfigMappingUtils.generateConfigClasses(ConfigMappingUtils.java:79)
	at io.quarkus.arc.deployment.ConfigBuildStep.generateConfigClasses(ConfigBuildStep.java:246)
```

### PR DESCRIPTION
**Pull request by angelozerr**
Validate that @ConfigMapping annotation can only be placed in interfaces

Fixes #424

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
I need to write test, but it should work:

image

image

**Comment by angelozerr** [comment]
@rgrunber you can review it now. Test is written.

File: quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/core/java/QuarkusConfigMappingASTVisitor.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		}
+		super.visit(node);
+		return true;
+	}
+
+	/**
+	 * Checks if the given type declaration annotated with @ConfigMapping annotation
+	 * is an interface.
+	 *
+	 * @param node       The method declaration to validate
```
Comment (rgrunber) [inline quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/core/java/QuarkusConfigMappingASTVisitor.java:79]: type declaration

File: quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/core/java/QuarkusConfigMappingASTVisitor.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			for (Object modifier : modifiers) {
+				if (modifier instanceof Annotation) {
+					Annotation annotation = (Annotation) modifier;
+					if (isMatchAnnotation(annotation, QuarkusConstants.CONFIG_MAPPING_ANNOTATION)) {
+						validateConfigMappingAnnotation(node, annotation);
+					}
+				}
+			}
+		} catch (JavaModelException e) {
+			LOGGER.log(Level.WARNING, "An exception occurred when attempting to validate the annotation marked method");
```
Comment (rgrunber) [inline quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/core/java/QuarkusConfigMappingASTVisitor.java:69]: marked type

**Comment by rgrunber** [comment]
I would just rebase on top of https://github.com/redhat-developer/quarkus-ls/pull/429 and then, feel free to merge.

**Comment by angelozerr** [comment]
Thanks @rgrunber for your review!

**Pull request merged**